swap/swap_test.go

```diff
@@ -682,6 +688,125 @@ func TestResetBalance(t *testing.T) {
+		t.Fatal(err)
+	}
+
+	cDummyPeer := newDummyPeerWithSpec(Spec)
+	dDummyPeer := newDummyPeerWithSpec(Spec)
+	creditor, err := debitorSwap.addPeer(cDummyPeer.Peer, creditorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+	debitor, err := creditorSwap.addPeer(dDummyPeer.Peer, debitorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// the creditor and debitor should start out with a balance of 0
+	if creditor.getBalance() != 0 {
+		t.Fatalf("expected balance to be 0 for peer %v, but it is %d", creditor.ID(), creditor.getBalance())
+	}
+	if debitor.getBalance() != 0 {
+		t.Fatalf("expected balance to be 0 for peer %v, but it is %d", debitor.ID(), debitor.getBalance())
+	}
+
+	// no cheques should be present at this point
+	if creditor.getLastSentCheque() != nil {
+		t.Fatalf("expected no cheque sent to peer %v, but it is %v", creditor.ID(), creditor.getLastSentCheque())
+	}
+	if debitor.getLastReceivedCheque() != nil {
+		t.Fatalf("expected no cheque received from peer %v, but it is %v", debitor.ID(), debitor.getLastReceivedCheque())
+	}
+
+	// set asymmetric balance and attempt to send cheque
+	if err = creditor.setBalance(-int64(ChequeDebtTolerance * 2)); err != nil {
+		t.Fatal(err)
+	}
+	// now simulate sending the cheque to the creditor from the debitor
+	if err = creditor.sendCheque(); err != nil {
+		t.Fatal(err)
+	}
+	cheque := creditor.getPendingCheque()
+
+	// simulate cheque handling
+	err = creditorSwap.handleEmitChequeMsg(ctx, debitor, &EmitChequeMsg{
+		Cheque: cheque,
+	})
+	// cheque should not have gone through as it would put the creditor in debt
+	if err == nil {
+		t.Fatal("expected invalid cheque reception to cause a failure, but got none")
+	}
+	// since the emission caused an error, the confirmation should not be simulated
+	// cheque should not be marked as sent, and should be pending
+	if creditor.getLastSentCheque() != nil {
+		t.Fatalf("expected no cheque sent to peer %v, but it is %v", creditor.ID(), creditor.getLastSentCheque())
+	}
+	if creditor.getPendingCheque() != cheque {
+		t.Fatalf("expected pending cheque for peer %v to be %v, but is %v", creditor.ID(), cheque, creditor.getPendingCheque())
+	}
+
+	// clear pending cheque to start over
+	if err = creditor.setPendingCheque(nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// set asymmetric balances to send a (barely) admissible cheque
+	if err = creditor.setBalance(-int64(ChequeDebtTolerance)); err != nil {
+		t.Fatal(err)
+	}
+
+	if err = creditor.sendCheque(); err != nil {
+		t.Fatal(err)
+	}
+	cheque = creditor.getPendingCheque()
+
+	// simulate cheque handling
+	err = creditorSwap.handleEmitChequeMsg(ctx, debitor, &EmitChequeMsg{
+		Cheque: cheque,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// simulate cheque confirmation
+	debitorSwap.handleConfirmChequeMsg(ctx, creditor, &ConfirmChequeMsg{
```

Comment:
this seems irrelevant to what this test is supposed to check.

Reply:
if i remove this, i have to skip the balances verification. 

cheque confirmation will cause the creditor reset its peer balance to 0.

i guess this isn't really what the test is about, so it can be skipped.

Reply:
removed

Reply:
see reply [here](https://github.com/ethersphere/swarm/pull/1983#issuecomment-576332679)